### DISCUSSION TITLE
Multipoint constraint

### DISCUSSION THREAD

**Discussion by ghost** [kinnala/scikit-fem#1101] [Q&A]
Hi, everyone) Have a nice weekend.
Is it possible to modify the mpc(...) function to support for eigenvalue problems?
Is there another way to solve eigenvalue problems with pbc using sckit-fem?

**Comment by kinnala** [comment]
You can try creating a periodic mesh, MeshTri1DG.init_tensor(..., periodic=[0]). It is lacking some features so be careful. Modifying mpc is not out of the question, I don't have the bandwidth right now but will review any PRs.

**Comment by ghost** [comment]
One more issue here so not to create a new discussion.
I have the following code:
```{python}
        m = MeshLine().refined(5)
        basis = Basis(m, ElementLineP1() * ElementLineP1())

        @BilinearForm
        def bilinf(u, T, v, V, w):
            return dot(grad(u), grad(v)) + 0.1 * grad(T)[0] * v + dot(grad(T), grad(V))

        K: spmatrix = bilinf.assemble(basis)
```
Using the debugger I found out that the args T and V are arrays of zeros. Is this how it should be or I did something wrong?

**Reply by kinnala** [reply]
Yes, one after another, (u, v) or (T, V) will be zeros. There used to be some optimizations in place to avoid evaluating everything with large zero arrays but they were removed to simplify other things.
